src/main/java/org/openjdk/jextract/impl/StructLayoutComputer.java

```diff
@@ -148,19 +140,26 @@ Declaration.Scoped finishRecord(String anonName) {
         } else if (anonName != null) {
             g = g.withName(anonName);
         }
-        Declaration.Scoped declaration = Declaration.struct(TreeMaker.CursorPosition.of(cursor), cursor.spelling(), g, fieldDecls.stream().toArray(Declaration[]::new));
+        Declaration.Scoped declaration = Declaration.struct(TreeMaker.CursorPosition.of(cursor), cursor.spelling(),
+                g, fieldDecls.stream().toArray(Declaration[]::new));
         return declaration;
     }
 
     // process bitfields if any and clear bitfield layouts
     private void handleBitfields() {
         if (bitfieldDecls != null) {
-            List<MemoryLayout> prevBitfieldLayouts = bitfieldLayouts;
             List<Declaration> prevBitfieldDecls = bitfieldDecls;
+            long prevBitfieldSize = bitfieldSize;
             bitfieldDecls = null;
+            bitfieldSize = 0;
             if (!prevBitfieldDecls.isEmpty()) {
-                addField(bitfield(prevBitfieldLayouts, prevBitfieldDecls.toArray(new Declaration.Variable[0])));
+                addField(offset, bitfield(prevBitfieldDecls.toArray(new Declaration.Variable[0])));
+            }
+            if (prevBitfieldSize == 0) {
```

Comment:
This `if` is new - there is a change in semantics from Java 20 to Java 21 - as Java 20 disallows creation of zero-length padding, while Java 21 allows it (which seems like a bug). So, to make jextract work against Java 20, this change was required.